### PR TITLE
 test(k8sprocessor): do not skip if other associations are already up to date

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Backporting https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14097

### CODE REVIEW

File: pkg/processor/k8sprocessor/kube/client.go [hunk 1]
```diff
@@ -490,24 +490,28 @@ func (c *WatchClient) addOrUpdatePod(pod *api_v1.Pod) {
...
+
 	if newPod.Name != "" && newPod.Namespace != "" {
-		c.Pods[generatePodIDFromName(newPod)] = newPod
+		identifiers = append(identifiers, generatePodIDFromName(newPod))
+	}
+
+	for _, identifier := range identifiers {
+		if identifier != "" {
+			// compare initial scheduled timestamp for existing pod and new pod with same IP
+			// and only replace old pod if scheduled time of new pod is newer? This should fix
```
Comment (kkujawa-sumo) [inline pkg/processor/k8sprocessor/kube/client.go:505]: Why is there a question mark after `newer`?

**Review by kkujawa-sumo — APPROVED** [review]

File: pkg/processor/k8sprocessor/kube/client.go (hunk 1 above)
Comment (sumo-drosiek) [reply pkg/processor/k8sprocessor/kube/client.go:505]: I think its typo

Comment (sumo-drosiek) [inline pkg/processor/k8sprocessor/kube/client.go:505]:
```suggestion
			// and only replace old pod if scheduled time of new pod is newer. This should fix
```

File: pkg/processor/k8sprocessor/kube/client.go [hunk 2]
```diff
@@ -490,24 +490,28 @@ func (c *WatchClient) addOrUpdatePod(pod *api_v1.Pod) {
...
+			// the case where scheduler has assigned the same IP to a new pod but update event for
+			// the old pod came in later.
```
Comment (sumo-drosiek) [inline pkg/processor/k8sprocessor/kube/client.go:507]:
```suggestion
			// compare initial scheduled timestamp for existing pod and new pod with same identifier
			// and only replace old pod if scheduled time of new pod is newer or equal.
			// This should fix the case where scheduler has assigned the same attributes (like IP address)
			// to a new pod but update event for the old pod came in later.
```

**Pull request merged**